functions/Backup-DbaDatabase.ps1

```diff
@@ -453,8 +453,17 @@ function Backup-DbaDatabase {
 
             if ($CompressBackup) {
                 if ($db.EncryptionEnabled) {
-                    Write-Message -Level Warning -Message "$dbName is enabled for encryption, will not compress"
-                    $backup.CompressionOption = 2
+                    $minVerForTDECompression = [version]'13.0.4446.0' # SQL Server 2016 CU 4
+                    $flagTDESQLVersion = $minVerForTDECompression -le $Server.version
+                    $flagTestBoundMaxTransferSize = test-bound 'MaxTransferSize'
```

Comment:
Why do you use Test-Bound and not just test for the value of $MaxTransferSize?

Reply:
And I'm a fan of nice case: test-bound -> Test-Bound

Reply:
I use test-bound in order to simulate the behavior described in the documentation that states that MaxTransferSize have to explicitly passed as parameter.
I corrected the nice case.